lm/kenlm.py

```diff
@@ -0,0 +1,138 @@
+import shutil
+
+from sisyphus import Job, Task, tk, gs
+
+from typing import Union, List, Optional
+import os
+import tempfile
+import subprocess as sp
+
+from i6_core.util import uopen
+
+
+class CompileKenLMJob(Job):
+    """
+    Compile KenLM and store a folder containing the binaries.
+
+    Please make sure the needed libraries (e.g. boost, zlib) are on your system or image
+    """
+
+    def __init__(self, repository: tk.Path):
+        """
+
+        :param repository: e.g. CloneGitRepositoryJob output for https://github.com/kpu/kenlm
+        """
+        self.repository = repository
+        self.out_binaries = self.output_path("kenlm_binary_folder")
+        self.rqmt = {"time": 0.5, "mem": 4, "cpu": 4}
+
+    def tasks(self):
+        yield Task("run", rqmt=self.rqmt)
+
+    def run(self):
+        with tempfile.TemporaryDirectory(prefix=gs.TMP_PREFIX) as td:
+            repo = os.path.join(td, "repo")
+            shutil.copytree(self.repository.get_path(), repo)
+            build_dir = os.path.join(repo, "build")
+            os.mkdir(build_dir)
+            sp.check_call(["cmake", ".."], cwd=build_dir)
+            sp.check_call(["make", "-j", str(self.rqmt["cpu"])], cwd=build_dir)
+            shutil.copytree(os.path.join(build_dir, "bin"), self.out_binaries.get_path())
+
+
+class KenLMplzJob(Job):
+    """
+    Run the lmplz command of the KenLM toolkit to create a gzip compressed ARPA-LM file
+    """
+
+    def __init__(
+        self,
+        text: Union[tk.Path, List[tk.Path]],
+        order: int,
+        interpolate_unigrams: bool,
+        pruning: Optional[List[int]],
+        vocabulary: Optional[tk.Path],
+        kenlm_binary_folder: tk.Path,
+    ):
+        """
+
+        :param text: training text data
+        :param order: "N"-order of the "N"-gram LM
+        :param interpolate_unigrams: Set True for KenLM default, and False for SRILM-compatibility.
+            Having this as False will increase the share of the unknown probability
+        :param pruning: absolute pruning threshold for each order,
+            e.g. to remove 3-gram and 4-gram singletons in a 4th order model use [0, 0, 1, 1]
+        :param vocabulary: a "single word per line" file to determine valid words,
+            everything else will be treated as unknown
+        :param kenlm_binary_folder: output of the CompileKenLMJob, or a direct link to the build
+            dir of the KenLM repo
+        """
+        self.text = text
+        self.order = order
+        self.interpolate_unigrams = interpolate_unigrams
+        self.pruning = pruning
+        self.vocabulary = vocabulary
+        self.kenlm_binary_folder = kenlm_binary_folder
+
+        self.out_lm = self.output_path("lm.gz")
+
+        self.rqmt = {"cpu": 1, "mem": 4.0, "time": 1}
```

Comment:
I mean you can always configure it later via job.rqmt, but yes I could add this as direct paramter.

Reply:
true.. my reasoning would be to put in the args of the job. otherwise, users will run once it crashes and then start thinking about it.. maybe not all users but my feeling/hope is that it would be less ^^